### ISSUE TITLE
debug with dotenv

### ISSUE THREAD

**Issue by TyIsI** [debug-js/debug#936]
Note: This is purely informational and I am documenting this so that it may help *someone*.

The TLDR snippet (in ESM/Typescript):
```
import Debug from 'debug'
import dotenv from 'dotenv'

// Save the current DEBUG environment variable if needed
const DEBUG = process.env.DEBUG

// Call Debug.disable() if needed/wanted
Debug.disable()

// Do dotenv things
dotenv.config({ path: 'my.env' })
dotenv.config({ path: 'my.env.development.local' })

// Parse environment
if (process.env.DEBUG != null) {
    // Update with new variable if set
    Debug.enable(process.env.DEBUG)
} else if (DEBUG != null) {
    // Fallback to previous DEBUG environment variable if it existed
    Debug.enable(DEBUG)
}
```

Quick background/facts:
- `enable` calls `save`, which saves the namespace argument into `process.env.DEBUG`.
- Don't call `disable`, because it will call `enable` with an empty string argument, causing `process.env.DEBUG` to be empty.
- But if  you want to make sure that `process.env.DEBUG` is clear, we need to make sure that we call disable *BEFORE* we do *anything* with the environment (i.e. `dotenv`).
- The ideal way to load the DEBUG variable would be to use `Debug.enable(Debug.load())`, but `Debug.load()` is not properly documented in the TS types and will cause validation errors.

I'll see if I can make a useful PR for this.

**Comment by Qix-** [comment]
I'm going to close this since it's not really an 'open' issue but if I see more issues filed w.r.t. dotenv in the future I'll pin this for visibility.

**Comment by TyIsI** [comment]
> Just my $0.02. Hardcoding anything about the environment into your source directly rarely works out for you in the long run.

In my case, I was trying to emulate the behaviour from the `create-react-app`, which I believe was based on the behaviour of the Ruby gem[1] that existed before the JS dotenv project came to exist.
It worked by trying to load different `.env` files according to the detected environment.

So in my example, the `// Do dotenv things` section would rather look something like this:
```Typescript
import { existsSync } from 'node:fs'
import { cwd } from 'node:process'
import path from 'node:path'

...

// Do dotenv things
const nodeEnv = process.env.NODE_ENV ?? 'development'

const envFiles = [`.env.${nodeEnv}.local`, '.env.local', `.env.${nodeEnv}`, '.env']
    .filter((envFile) => existsSync(path.resolve(path.join(cwd(),envFile))))

for(const envFile of envFiles) {
    dotenv.config({ path: envFile })
}
```

As an example scenario, it would allow me to easily switch or (selectively) override (`debug`) settings.
Which could handy in case you have a generic .env file included in a Docker build and want to override only particular variables in a specific environment, or want or need to tune them. This is a safe(r) escape hatch without having to manage full environment variables and the potential discrepancies this might bring. (Which is obviously why `dotenv-vault` came to rise.)

Though, you are absolutely correct, this would ideally be implemented using a preload script that provides this functionality.

[1]: https://github.com/bkeepers/dotenv
